src/yunohost/certificate.py

```diff
@@ -837,7 +843,10 @@ def _dns_ip_match_public_ip(public_ip, domain):
 
 def _domain_is_accessible_through_HTTP(ip, domain):
     try:
-        requests.head("http://" + ip, headers={"Host": domain})
+        requests.head("http://" + ip, headers={"Host": domain}, timeout=30)
+    except Timeout as e:
+        # XXX what should we do here? retry?
```

Comment:
@julienmalik Well yep, but @Psycojoker was pushing to have a 30 sec timeout 😛 . But I'd be in favor of this too.